components/Modals/BatchTransfer/index.tsx

```diff
@@ -0,0 +1,343 @@
+import { useStore } from 'effector-react'
+import {
+    $modalTransfer,
+    updateModalTx,
+    updateModalTxMinimized,
+    updateTransferModal,
+} from '../../../src/store/modal'
+import CloseReg from '../../../src/assets/icons/ic_cross.svg'
+import CloseBlack from '../../../src/assets/icons/ic_cross_black.svg'
+import stylesDark from './styles.module.scss'
+import stylesLight from './styleslight.module.scss'
+import Image from 'next/image'
+import { useEffect, useState } from 'react'
+import * as tyron from 'tyron'
+import { useDispatch, useSelector } from 'react-redux'
+import { RootState } from '../../../src/app/reducers'
+import Spinner from '../../Spinner'
+import { Donate, Selector } from '../..'
+import { useTranslation } from 'next-i18next'
+import smartContract from '../../../src/utils/smartContract'
+import routerHook from '../../../src/hooks/router'
+import { $resolvedInfo } from '../../../src/store/resolvedInfo'
+import Tydra from '../../../src/assets/logos/tydra.json'
+import arweave from '../../../src/config/arweave'
+import { ZilPayBase } from '../../ZilPay/zilpay-base'
+import { setTxId, setTxStatusLoading } from '../../../src/app/actions'
+import ThreeDots from '../../Spinner/ThreeDots'
+import CloseIcoReg from '../../../src/assets/icons/ic_cross.svg'
+import CloseIcoBlack from '../../../src/assets/icons/ic_cross_black.svg'
+import ContinueArrow from '../../../src/assets/icons/continue_arrow.svg'
+import TickIco from '../../../src/assets/icons/tick.svg'
+import { $donation, updateDonation } from '../../../src/store/donation'
+import { TransitionParams } from 'tyron/dist/blockchain/tyronzil'
+import { toast } from 'react-toastify'
+import toastTheme from '../../../src/hooks/toastTheme'
+
+function Component() {
+    const { t } = useTranslation()
+    const { getSmartContract } = smartContract()
+    const { navigate } = routerHook()
+    const dispatch = useDispatch()
+    const net = useSelector((state: RootState) => state.modal.net)
+    const loginInfo = useSelector((state: RootState) => state.modal)
+    const modalTransfer = useStore($modalTransfer)
+    const resolvedInfo = useStore($resolvedInfo)
+    const donation = useStore($donation)
+    const isLight = useSelector((state: RootState) => state.modal.isLight)
+    const styles = isLight ? stylesLight : stylesDark
+    const Close = isLight ? CloseBlack : CloseReg
+    const CloseIco = isLight ? CloseIcoBlack : CloseIcoReg
+
+    const [selectedCoin, setSelectedCoin] = useState<any>([])
+    const [inputCoin, setInputCoin] = useState<any>([])
+    const [savedCurrency, setSavedCurrency] = useState(false)
+    const [isLoading, setIsLoading] = useState(false)
+
+    const outerClose = () => {
```

Comment:
shall me move this & use globally?

Reply:
hmmm, I don't think so. Since it's short and quite simple